### PR TITLE
BLoC

### PR DESCRIPTION
**Pull request by dmitry-rachko-idf**
implemented a bloc with a streambuilder

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -36,6 +36,9 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
+  flutter_bloc: ^8.0.1
```
Comment (andrei-lastochkin-idf) [inline pubspec.yaml]: remove unnecessary imports

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -48,3 +44,7 @@ app.*.map.json
 /android/app/debug
 /android/app/profile
 /android/app/release
+packages/presentation/pubspec.lock
+packages/domain/pubspec.lock
+pubspec.lock
```
Comment (sergei-mikhailovskii-idf) [inline .gitignore:49]: seems that u can do smth like this */pubspec.lock, to ignore all pubspecs

File: packages/domain/lib/palindrome_usecase.dart [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class PalindromeUseCase {
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/lib/palindrome_usecase.dart:1]: in bigger projects this class should be in separate package, but now it's ok

File: packages/presentation/lib/home_page.dart [hunk 4]
```diff
@@ -12,82 +13,82 @@ class MyHomePage extends StatefulWidget {
...
+                return Container(
+                  color: Colors.red,
+                );
+              }
+            } else {
+              return Container(
+                color: Colors.green,
+              );
+            }
+          }),
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/home_page.dart:55]: in the bigger project stream builder should be encapsulated in a separate widget as I showed u. but now it's ok

File: pubspec.lock [hunk 5]
```diff
@@ -8,6 +8,13 @@ packages:
       url: "https://pub.dartlang.org"
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.lock:8]: gitignore

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**